easy-rules-core/src/main/java/org/jeasy/rules/api/Rules.java

```diff
@@ -65,29 +65,43 @@ public Rules(Rule... rules) {
      * @param rules to register
      */
     public Rules(Object... rules) {
-        for (Object rule : rules) {
-            this.register(rule);
-        }
+        this.register(rules);
     }
 
     /**
      * Register a new rule.
      *
-     * @param rule to register, must not be null
+     * @param rules to register, must not be null
      */
-    public void register(Object rule) {
-        Objects.requireNonNull(rule);
-        rules.add(RuleProxy.asRule(rule));
+    public void register(Object... rules) {
+        Objects.requireNonNull(rules);
+        for (Object rule : rules) {
+            Objects.requireNonNull(rule);
+            this.rules.add(RuleProxy.asRule(rule));
+        }
     }
 
     /**
      * Unregister a rule.
      *
-     * @param rule to unregister, must not be null
+     * @param rules to unregister, must not be null
+     */
+    public void unregister(Object... rules) {
+        Objects.requireNonNull(rules);
+        for (Object rule : rules) {
+            Objects.requireNonNull(rule);
+            this.rules.remove(RuleProxy.asRule(rule));
+        }
+    }
+
+    /**
+     * Unregister a set of rules.
+     *
+     * @param rulesSet set to unregister, must not be null
      */
-    public void unregister(Object rule) {
-        Objects.requireNonNull(rule);
-        rules.remove(RuleProxy.asRule(rule));
+    public void unregisterAll(Set<?> rulesSet) {
```

Comment:
I thought we agreed to not duplicate `umregister`/`unregsiterAll` and have a single `unregister` method with varargs. I will remove this on merge.